### PR TITLE
Inspect patterns on Red Hat

### PR DESCRIPTION
**Pull request by Werni**

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged be2c86736fd9a85782c8179e3b3a57231877d572 into 505dfe4042ed8d49efe39d4deb5426506d6b4a60 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 1bf7bbb06f3f87effbc566818c44a2b0132af3ea into 505dfe4042ed8d49efe39d4deb5426506d6b4a60 )

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
http://ci.opensuse.org/job/machinery-prophet/225644/console
If the given link has expired, you can force a Prophet rerun by just deleting this comment.
( Failure: Merged 5f601285052693dd310d78d052221f93bddf3aed into 505dfe4042ed8d49efe39d4deb5426506d6b4a60 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 159853a84eee55854a26dd9e0e245e6afc818060 into 505dfe4042ed8d49efe39d4deb5426506d6b4a60 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged cde65ed46cb669cc74315f91130d6c91ec0ef92b into 505dfe4042ed8d49efe39d4deb5426506d6b4a60 )

File: spec/unit/patterns_inspector_spec.rb [hunk 1]
```diff
@@ -73,6 +74,9 @@
     end
 
     it "returns an empty array when there are no patterns installed" do
+      allow(system).to receive(:has_command?).with("zypper").and_return(true)
+      allow(system).to receive(:has_command?).with("yum").and_return(false)
```
Comment (thardeck) [inline spec/unit/patterns_inspector_spec.rb]: this can be removed

File: spec/unit/patterns_inspector_spec.rb [hunk 2]
```diff
@@ -88,6 +95,9 @@
     end
 
     it "raises an error if zypper is locked" do
+      allow(system).to receive(:has_command?).with("zypper").and_return(true)
+      allow(system).to receive(:has_command?).with("yum").and_return(false)
+
```
Comment (thardeck) [inline spec/unit/patterns_inspector_spec.rb]:
```
      allow(system).to receive(:has_command?).with("zypper").and_return(true)
      allow(system).to receive(:has_command?).with("yum").and_return(false)
```
I would move this to a before block.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 92d74a800f22e0f05174c6bc609dd5dc84cf1694 into 505dfe4042ed8d49efe39d4deb5426506d6b4a60 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged c808441e2c1a668a5e31dce666fe5119e15acd58 into 505dfe4042ed8d49efe39d4deb5426506d6b4a60 )

**Comment by thardeck** [comment]
lgtm

**Pull request merged**